tinygrad/helpers.py

```diff
@@ -1,18 +1,23 @@
 from __future__ import annotations
-import platform
-from dataclasses import dataclass, asdict
-import os, math, functools, time, re
+import os, functools, platform, time
+from weakref import KeyedRef, ref
+from _weakref import _remove_dead_weakref
 import numpy as np
-from typing import Tuple, Union, List, NamedTuple, Final, Iterator, ClassVar, Optional, Callable, Any
+from typing import Dict, Tuple, Union, List, NamedTuple, Final, Iterator, ClassVar, Optional, Callable, Any
+from math import prod
+
 ShapeType = Tuple[int, ...]
 # NOTE: helpers is not allowed to import from anything else in tinygrad
 OSX = platform.system() == "Darwin"
 
 def dedup(x): return list(dict.fromkeys(x))   # retains list order
-def prod(x:Union[List[int], Tuple[int, ...]]) -> int: return math.prod(x)
-def argfix(*x): return tuple() if len(x) == 0 else tuple(x[0]) if isinstance(x[0], (tuple, list)) else tuple(x)
+def argfix(*x): 
+  if x[0].__class__ in {tuple, list}:
+    try: return tuple(x[0])
+    except IndexError: return tuple()
```

Comment:
No way an exception is faster than a len check

Reply:
This is a statistical optimization, `try` blocks that don't throw an exception are virtually free and in the overwhelming majority of calls to argfix we are in the exception-free path